azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/implementation/FlowLogSettingsImpl.java

```diff
@@ -0,0 +1,156 @@
+        super(inner);
+        this.parent = parent;
+        this.nsgId = nsgId;
+    }
+
+    @Override
+    public FlowLogSettings apply() {
+        return applyAsync().toBlocking().last();
+    }
+
+    @Override
+    public Observable<FlowLogSettings> applyAsync() {
+        return this.parent().manager().inner().networkWatchers()
+                .setFlowLogConfigurationAsync(parent().resourceGroupName(), parent().name(), this.inner())
+                .map(new Func1<FlowLogInformationInner, FlowLogSettings>() {
+            @Override
+            public FlowLogSettings call(FlowLogInformationInner flowLogInformationInner) {
+                return new FlowLogSettingsImpl(FlowLogSettingsImpl.this.parent, flowLogInformationInner, nsgId);
+            }
+        });
+    }
+
+    @Override
+    public ServiceFuture<FlowLogSettings> applyAsync(ServiceCallback<FlowLogSettings> callback) {
+        return ServiceFuture.fromBody(applyAsync(), callback);
+    }
+
+    @Override
+    public Update withLogging() {
+        this.inner().withEnabled(true);
+        return this;
+    }
+
+    @Override
+    public Update withoutLogging() {
+        this.inner().withEnabled(false);
+        return this;
+    }
+
+    @Override
+    public Update withStorageAccount(String storageId) {
+        this.inner().withStorageId(storageId);
+        return this;
+    }
+
+    @Override
+    public Update withRetentionPolicyEnabled() {
+        ensureRetentionPolicy();
+        this.inner().retentionPolicy().withEnabled(true);
+        return this;
+    }
+
+    @Override
+    public Update withRetentionPolicyDisabled() {
+        ensureRetentionPolicy();
+        this.inner().retentionPolicy().withEnabled(false);
+        return this;
+    }
+
+    @Override
+    public Update withRetentionPolicyDays(int days) {
+        ensureRetentionPolicy();
+        this.inner().retentionPolicy().withDays(days);
+        return this;
+    }
+
+    private void ensureRetentionPolicy() {
+        if (this.inner().retentionPolicy() == null) {
+            this.inner().withRetentionPolicy(new RetentionPolicyParameters());
+        }
+    }
+
+    @Override
+    public Update update() {
+        return this;
+    }
+
+    @Override
+    protected Observable<FlowLogInformationInner> getInnerAsync() {
+        return this.parent().manager().inner().networkWatchers()
+                .getFlowLogStatusAsync(parent().resourceGroupName(), parent().name(), inner().targetResourceId());
+    }
+
+    @Override
+    public NetworkWatcherImpl parent() {
+        return parent;
+    }
+
+    @Override
+    public String key() {
+        return null;
+    }
+
+    @Override
+    public String targetResourceId() {
+        return inner().targetResourceId();
+    }
+
+    @Override
+    public String storageId() {
+        return inner().storageId();
+    }
+
+    @Override
+    public boolean enabled() {
+        return inner().enabled();
```

Comment:
is inner().enabled is a Boolean? if so if it is null then invoking this getter may throw error right? you can use `Utils.toPrimitiveBoolean(inner().enabled());`